system/Database/Query.php

```diff
@@ -406,10 +404,15 @@ public function debugToolbarDisplay(): string
 
         $sql = $this->finalQueryString;
 
-        foreach ($highlight as $term) {
-            $sql = str_replace($term, '<strong>' . $term . '</strong>', $sql);
-        }
+        $search = '/\b(?:' . implode('|', $highlightWords) . ')\b/';
+
+        $sql = preg_replace_callback($search, static function ($matches) {
+            return '<strong>' . str_replace(' ', '&nbsp;', $matches[0]) . '</strong>';
+        }, $sql);
+
+        $sql = str_replace(['(', ')'], ['<strong>(</strong>', '<strong>)</strong>'], $sql);
 
+        /** @noRector \Rector\CodeQuality\Rector\Return_\SimplifyUselessVariableRector */
```

Comment:
I would agree with Rector here to simplify the return. Besides, our coding standard also includes this rule to simplify the return when possible (although it is not taking effect here).